### PR TITLE
[Bugfix] Use re2 when hyperscan report errors

### LINKED ISSUE #10364: Report error with like/regexp when pattern is larger 16000
### LIKE
```
pattern = a%a%bc123|aaaaaaaaaaaaaaaaa....aaa;
```
pattern's size is larger than 16000.
```
select like ("abc123", pattern);
```

### REGEXP
```
pattern = a.aaaaaaaaaaaaaaaaa....aaa;
```
pattern's size is larger than 16000.
```
select regexp ("abc123", pattern);
````
### Real behavior (Required)

ERROR 1064 (HY000): Invalid regex expression:
"Pattern length exceeds limit" is ouput in be.INFO

### PR DESCRIPTION
**Pull request by Pslydhh**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixed https://github.com/StarRocks/starrocks/issues/10364

We use re2 instead of hyperscan when hyperscan report errors.

### CODE REVIEW

File: be/src/exprs/vectorized/like_predicate.h [hunk 1]
```diff
@@ -134,6 +136,10 @@ class LikePredicate {
 
     static ColumnPtr regex_match_partial(FunctionContext* context, const Columns& columns);
 
+    static const size_t MAX_PATTERN_OF_HYPERSCAN = 16000;
```
Comment (Seaven) [inline be/src/exprs/vectorized/like_predicate.h:139]: move to .cpp

Comment (Pslydhh) [reply be/src/exprs/vectorized/like_predicate.h]: done.

**Review by kangkaisen — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/exprs/vectorized/like_predicate.cpp [hunk 2]
```diff
@@ -190,6 +224,44 @@ ColumnPtr LikePredicate::regex_fn(FunctionContext* context, const Columns& colum
...
+    ColumnBuilder<TYPE_BOOLEAN> result(num_rows);
+
+    for (int row = 0; row < num_rows; ++row) {
+        if (value_viewer.is_null(row)) {
+            result.append_null();
+            continue;
+        }
+
+        bool v = false;
+        if constexpr (like) {
```
Comment (imay) [inline be/src/exprs/vectorized/like_predicate.cpp:252]: there is a function whose name is like, you'd better to give this like another name.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by kangkaisen — DISMISSED** [review]

File: be/src/exprs/vectorized/like_predicate.cpp (hunk 2 above)
Comment (Pslydhh) [reply be/src/exprs/vectorized/like_predicate.cpp]: > there is a function whose name is like, you'd better to give this like another name.

Have fixed

**Comment by Pslydhh** [comment]
These PR needs to be modified. Don't merge it.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**